`Feature` SRS (EXPOSUREAPP-13454)

mtwalli opened the pull request:
👷🏼‍♂️ 🚧 
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13454

mtwalli committed:
Delete unused

mtwalli committed:
Setup SRS skeleton

mtwalli committed:
Update Module

mtwalli committed:
Update name

mtwalli committed:
Merge branch 'release/2.29.x' into feature/13454-srs

Ein-Tim commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/viewmodel/SubmissionDispatcherViewModel.kt:
```diff
@@ -40,6 +46,26 @@ class SubmissionDispatcherViewModel @AssistedInject constructor(
+    fun onSelfTestClicked() = launch {
+        try {
+            srsLocalChecker.check()
+            // proceed with SRS type  param
```
Double whitespace
```suggestion
            // proceed with SRS type param
```

Ein-Tim reviewed (commented):
One comment.

mtwalli committed:
Update Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/viewmodel/SubmissionDispatcherViewModel.kt

Co-authored-by: Tim <67682506+Ein-Tim@users.noreply.github.com>

mtwalli committed:
Fix test

mtwalli committed:
Merge branch 'release/2.29.x' into feature/13454-srs

Corona-Warn-App-Technical-User committed:
Protocol Buffer update for release 2.29.x at Wed Nov  2 15:16:40 UTC 2022 (#5679)

* Update from https://github.com/corona-warn-app/cwa-protocol-buffers/commit/a506cf63fc594b148ff67353dbf7d5315727d143

* Bump to next Version: 2.29.0.1 (#5680)

Co-authored-by: Mohamed <mohamed.metwalli@sap.com>

mtwalli committed:
Srs settings (EXPOSUREAPP-14186) (#5678)

* SRS Settings

* Lint

mtwalli committed:
Self Report Submission config param (EXPOSUREAPP-14187) (#5681)

* Self Report Submission config param

* Lint

* Fix test

* Local prerequisites (EXPOSUREAPP-14185) (#5682)

* Update SrsLocalChecker.kt

* Check duration in hours

* Time since submission

* Create SrsLocalCheckerTest.kt

* lint

SamuraiKek committed:
Update Register Test Card in Home Fragment for SRS (EXPOSUREAPP-14173) (#5683)

* Update homescreen card with the new srs UI.

* Update button string resource.

* Delete old strings.

* Use unique IDs

* Fix drawable issue for Android 6.0.

mtwalli committed:
Merge branch 'release/2.29.x' into feature/13454-srs

schauersbergern committed:
Srs submission consent screen (EXPOSUREAPP-14176) (#5686)

* Add SrsSubmissionConsentFragment

* add navigation

* add newlines rename package

* remove consent dialog

* move legal strings to legal string file

* Add SrsSubmissionConsentFragmentTest
Add key retrieval to SrsSubmissionConsentFragmentViewModel

* remove nullable param

* remove duplicated bullet point layout

* Handle no tracing enabled

* remove duplicated bullet point layout
changed navargs naming

* fix test

* one param

* mock view model

* Deeplink

* Update SrsSubmissionConsentFragmentViewModel.kt

* Fix

* Adjust margins

* Fix layout

* fix layout

Co-authored-by: Mohamed Metwalli <mohamed.metwalli@sap.com>

mtwalli committed:
Merge branch 'release/2.29.x' into feature/13454-srs

SamuraiKek committed:
Update test dispatcher screen (EXPOSUREAPP-14174) (#5689)

* Adapt dispatcher screen and add custom dispatcher card view.

* Add strings to DE file.

* Move layout to main.

* Link screen

* One click handler

* Add navigation to consent fragment.

* Make tan cards visible

Co-authored-by: Mohamed Metwalli <mohamed.metwalli@sap.com>

mtwalli committed:
SRS Server Api (EXPOSUREAPP-14188, 14189) (#5687)

* Sever / API

* Lint

* Srs Device attestation

* Set steps

* Test / Run

* Lint

* lint

* Logs

* Auth response / error codes

* Submission error codes

* Split

* Revert Jackson converter

* Refactor attestation

* Map safety net errors

* Test

* Tests

* otp flow

* Test Result

* Log

* Update title

* Save last submission time

* Update SubmissionTaskTest.kt

* Update SrsSubmissionSettingsTest.kt

* Add test

* Update SrsSubmissionRepositoryTest.kt

* Device time

* Display submission time

* Extend test menu

* Cover reset

* Fix

* Check error code first

* Create SrsPlaybookTest.kt

* Cover OTP error

* Lint

* Create SrsSubmissionServerTest.kt

* Update SrsSubmissionServerTest.kt

* add new options

SamuraiKek committed:
Remove tan tiles from dispatcher screen (DEV) (#5692)

* Remove TAN tiles from dispatcher screen UI.

* Fix test.

mtwalli committed:
Merge branch 'release/3.0.x' into feature/13454-srs

schauersbergern committed:
Srs test type selection fragment (EXPOSUREAPP-14177) (#5691)

* add SrsTypeSelectionFragment

* refactor to recyclerview
add navigation to symptoms

* remove arguments

* update SrsSubmissionType

* Delegate selection to viewmodel / checked text

* Recycler

* Remove layout root
Add CommonCloseDialog.kt
Add Screenshot test

* Remove CommonCloseDialog.kt

Co-authored-by: Mohamed Metwalli <mohamed.metwalli@sap.com>

mtwalli committed:
Update strings.xml (#5696)

mtwalli committed:
Dev menu (EXPOSUREAPP-14182) (#5695)

* Extend SRS dev menu

* Bind Dev SRS settings

* Device Time State

* Force update config

* Fix Tests

* Update SrsAuthorizationServerTest.kt

* AppConfig sources

* Lint

* Delete

* Rename

* Create DefaultSrsDevSettingsTest.kt

* lint

MikeMcC399 commented on Corona-Warn-App/src/main/res/values/srs_submission_strings.xml:
```diff
@@ -0,0 +1,89 @@
+    <!-- XHED: Submission consent headline -->
+    <string name="srs_submission_consent_headline_text">Andere warnen</string>
+    <!-- XHED: Submission consent disclaimer sub headline -->
+    <string name="srs_submission_consent_disclaimer_subheadline_text">Bevor Sie andere warnen können, ist ihr Einverständnis erforderlich.</string>
```
"ihr Einverständnis" => "Ihr Einverständnis"

MikeMcC399 reviewed (changes requested):
Typo - see in-line comment

mtwalli committed:
Submissions in  journal (EXPOSUREAPP-14181) (#5693)

* Sever / API

* Lint

* Srs Device attestation

* Set steps

* Test / Run

* Lint

* lint

* Logs

* Auth response / error codes

* Submission error codes

* Split

* Revert Jackson converter

* Refactor attestation

* Map safety net errors

* Test

* Tests

* otp flow

* Test Result

* Log

* Update title

* Save last submission time

* Update SubmissionTaskTest.kt

* Update SrsSubmissionSettingsTest.kt

* Add test

* Update SrsSubmissionRepositoryTest.kt

* Device time

* Display submission time

* Extend test menu

* Cover reset

* Fix

* Check error code first

* Create SrsPlaybookTest.kt

* Cover OTP error

* Lint

* Create SrsSubmissionServerTest.kt

* Update SrsSubmissionServerTest.kt

* add new options

* Extend journal

* Report submission

* lint

* Logs

* Clean

* style

* Clean obsolete submissions in the diary

* Update ContactDiaryDataRetentionCalculationTest.kt

* Reusable

* Create SubmissionReporterTest.kt

* Update ContactDiaryOverviewViewModelTest.kt

* Screenshots Test

* Update SrsSubmissionRepositoryTest.kt

* DB test

* Update SrsSubmissionRepositoryTest.kt

* Update ContactDiaryOverviewFragmentTest.kt

MikeMcC399 commented on Corona-Warn-App/src/main/res/values-de/srs_submission_strings.xml:
```diff
@@ -0,0 +1,90 @@
+    <!-- XHED: Submission consent headline -->
+    <string name="srs_submission_consent_headline_text">Andere warnen</string>
+    <!-- XHED: Submission consent disclaimer sub headline -->
+    <string name="srs_submission_consent_disclaimer_subheadline_text">Bevor Sie andere warnen können, ist ihr Einverständnis erforderlich.</string>
```
"ihr Einverständnis"?

mtwalli committed:
Merge branch 'release/3.0.x' into feature/13454-srs

mtwalli committed:
SRS  errors (EXPOSUREAPP-14175) (#5701)

* url

* ErrorCode messages

* Update SrsAuthorizationServerTest.kt

* Title

* No wild card

* Error codes

mtwalli committed:
Move Android ID to payload (DEV) (#5703)

* Move Android ID to payload

* Import right proto buf

* lint

schauersbergern committed:
Update thank you screen (EXPOSUREAPP-14180) (#5700)

* add contact entry
add screenshot test

* remove unused id

SamuraiKek committed:
SRS Symptoms flow (EXPOSUREAPP-14179) (#5699)

* Add symptoms intro flow.

* Change type to self test.

* Fix strings.

* Fix srs_nav_graph.xml

* Add type to navigation elements.

* Fix cancel dialog

* Display right dialog.

* klint.

Co-authored-by: Mohamed Metwalli <mohamed.metwalli@sap.com>

mtwalli committed:
Srs Flow Tweaking (EXPOSUREAPP-14184) (#5704)

* Add Submission done screen

* Tweak flow

* display error code

* Navigation

* Lint

* Handle back

* Lint

mtwalli committed:
Typo

mtwalli commented:
> Typo - see in-line comment

Fixed

MikeMcC399 commented on Corona-Warn-App/src/main/res/values-de/srs_submission_strings.xml:
Is `\n\Wenn` correct?
Previously it was `\n\nWenn` and it looks like one `\n` was intended to be deleted.

Same issue in file `values` directory as well.

mtwalli replied:
https://github.com/corona-warn-app/cwa-app-android/pull/5709
